NDTensors/src/Unwrap/test/runtests.jl

```diff
@@ -138,4 +138,33 @@ include("../../../test/device_list.jl")
   x = dev(randn(elt, 4, 4))
   permutedims!(expose(y), expose(x), (2, 1))
   @test NDTensors.cpu(y) == transpose(NDTensors.cpu(x))
+
+  ##########################################
+  ### Testing an issue with CUDA&Metal transpose/adjoint mul
+  A = dev(randn(Float64, (3, 2)))
+  B = dev(randn(Float64, (3, 4)))
+  C = dev(randn(Float64, (4, 2)))
+  Cp = copy(C)
+  if (dev == NDTensors.cu)
+    CUDA.allowscalar(false)
+  end
+  ## This fails with scalar indexing 
+  #mul!(transpose(C), transpose(A), B, 1.0, 0.0)
+  mul!(C, transpose(B), A, 1.0, 0.0)
+  mul!(expose(transpose(Cp)), expose(transpose(A)), expose(B), 1.0, 0.0)
+  @test C ≈ Cp
+  Cp = fill!(similar(C), 0.0)
+  ## Try calling mul!! with transposes to verify that code works
+  NDTensors.mul!!(transpose(Cp), transpose(A), B, 1.0, 0.0)
+  @test C ≈ Cp
+
+  Cp = fill!(similar(C), 0.0)
+  ## This fails with scalar indexing 
+  #mul!(C', A', B, 1.0, 0)
+  mul!(C, B', A, 1.0, 0.0)
```

Comment:
```suggestion
  mul!(C, B', A, true, false)
```
Again, to be more agnostic about the element type (same changes below). `1.0` and `0.0` are `Float64`.

Reply:
I made this fix